lib/cartopy/io/ogc_clients.py

```diff
@@ -232,8 +232,8 @@ def _image_and_extent(self, wms_proj, wms_srs, wms_extent, output_proj,
         return LocatedImage(wms_image, extent)
 
     def fetch_raster(self, projection, extent, target_resolution):
-        service = self.service
         min_x, max_x, min_y, max_y = extent
+        target_resolution = [np.ceil(val) for val in target_resolution]
```

Comment:
OK right. I assumed that the ceiling on the `min_x` would actually make it smaller, like in the test you say min is 19.1 and you get 20. Wouldn't you want to treat each end differently? Or maybe I misunderstand this!